src/cogent3/core/new_sequence.py

```diff
@@ -1113,34 +1117,51 @@ def add_feature(
             feature_data.pop(discard)
         return self.make_feature(feature_data)
 
-    def to_moltype(self, moltype):
+    def to_moltype(self, moltype: str):
         """returns copy of self with moltype seq
 
         Parameters
         ----------
-        moltype : str
+        moltype
             molecular type
+
+        Notes
+        -----
+        This method cannot convert between nucleic acids and proteins. Use
+        get_translation() for that.
         """
         from cogent3.core import new_moltype
 
         if not moltype:
             raise ValueError(f"unknown moltype '{moltype}'")
 
         moltype = new_moltype.get_moltype(moltype)
+
         if moltype is self.moltype:
             return self
 
-        s = moltype.coerce_str(self._seq.str_value)
-        moltype.verify_sequence(s, gaps_allowed=True, wildcards_allowed=True)
-        sv = SeqView(seq=s, alphabet=moltype.most_degen_alphabet())
-        new = moltype.make_seq(sv, name=self.name, info=self.info)
+        if len(self.moltype.alphabet) == len(moltype.alphabet):
+            # converting between dna and rna
+            seq = (
+                moltype.most_degen_alphabet()
+                .array_to_bytes(array(self))
+                .decode("utf-8")
+            )
+
+        else:
+            # assume converting between ASCII/BYTES and nucleic acid
+            seq = str(self)
+
+        if not moltype.most_degen_alphabet().is_valid(seq):
+            raise ValueError(
+                f"Changing from old moltype={self.moltype.label} to new "
+                f"moltype={moltype.label} is not valid for this data"
+            )
+        sv = SeqView(seq=seq, alphabet=moltype.most_degen_alphabet())
```

Comment:
add a comment to the docstring that if this method is applied to a sequence that is part of a sequence collection, the result is no longer part of that collection.